src/config.c

```diff
@@ -2375,6 +2376,17 @@ static int isValidShutdownOnSigFlags(int val, const char **err) {
     return 1;
 }
 
+static int isValidAuxString(char *val,const char **err) {
+    for (unsigned i = 0; i < strlen(val); i++) {
+        if (!(isalnum(val[i]) || (strchr("!#$%&()*+:;<>?@[]^{|}~", val[i]) == NULL))){
+	    *err = "Human announced Nodename may only contain alphanumeric characters, "
+                "hyphens, dots or underscore";
+            return 0;
+	}
+    }
+    return 1;
+}
```

Comment:
Let's update the function in cluster.c to accept char * + length so this can become:
```suggestion
static int isValidAuxString(char *val,const char **err) 
    if (isValidAuxString(val, strlen(val)) {
        *err = "Human announced Nodename may only contain alphanumeric characters, "
                "hyphens, dots or underscore";
        return 0;
    }
    return 1;
}
```

Reply:
I have updated isValidAuxString function in cluster.c, so that it can be used in config.c